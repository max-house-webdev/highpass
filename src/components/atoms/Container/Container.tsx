import { ReactNode } from "react";

import { Container as ChakraContainer, ContainerProps } from "@chakra-ui/react";

export interface IContainerProps extends ContainerProps {
  children: ReactNode;
}

export function Container(props: IContainerProps) {
  const { children, ...rest } = props;

  return (
    <ChakraContainer
      data-testid="Container"
      variant={"mainContainer"}
      {...rest}
    >
      {children}
    </ChakraContainer>
  );
}
