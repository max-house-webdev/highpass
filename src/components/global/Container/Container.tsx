import { Container as ChakraContainer, ContainerProps } from "@chakra-ui/react";
import { ReactNode } from "react";

export interface IContainerProps extends ContainerProps {
  children: ReactNode;
}

export function Container(props: IContainerProps) {
  const { children, ...rest } = props;

  return (
    <ChakraContainer
      data-testid="Container"
      variant={"mainContainer"}
      {...rest}
    >
      {children}
    </ChakraContainer>
  );
}
