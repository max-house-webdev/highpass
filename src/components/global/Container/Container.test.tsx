import React from "react";

import { render, screen } from "@testing-library/react";
import "@testing-library/jest-dom/extend-expect";

import renderer from "react-test-renderer";

import { Container } from "./Container";

describe("<Container />", () => {
  test("it should mount", () => {
    const Children = () => <div data-testid="children">children</div>;

    render(
      <Container>
        <Children />
      </Container>
    );

    const container = screen.getByTestId("Container");

    expect(container).toBeInTheDocument();

    const children = screen.getByTestId("children");
    expect(container).toContainElement(children);
    expect(children).toHaveTextContent("children");
  });
});

test("<Container /> renders correctly", () => {
  const tree = renderer
    .create(
      <Container>
        <div>children</div>
      </Container>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
