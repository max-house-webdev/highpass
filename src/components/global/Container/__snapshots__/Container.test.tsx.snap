// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Container /> renders correctly 1`] = `
<div
  className="chakra-container css-0"
  data-testid="Container"
>
  <div>
    children
  </div>
</div>
`;
